### PR TITLE
[releasenotes] Clarify purpose and usage of the `Upgrade Notes` section

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Clarifies the purpose and the usage of the `Upgrade Notes` section.

### Motivation

Allow devs to understand what this section is for. Notes in this section should remain rare so that it's easy for a user to quickly know what they need to pay attention to before upgrading.

### CODE REVIEW

**Review by gbbr — DISMISSED** [review]
Thank you

File: docs/dev/contributing.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ users and not its developers.
       collection, `kube_service` tagging) is not implemented
   ```
 
-- `upgrade`: List action to take or limitation that could arise upon upgrading the agent.
+- `upgrade`: List actions to take or limitations that could arise upon upgrading the agent. Notes here must include steps that users can follow to 1. know if they're affected and 2. handle the change gracefully on their end.
```
Comment (ruthnaebeck) [inline docs/dev/contributing.md]:
```suggestion
- `upgrade`: List actions to take or limitations that could arise upon upgrading the Agent. Notes here must include steps that users can follow to 1. know if they're affected and 2. handle the change gracefully on their end.
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]
Small change

**Review by ruthnaebeck — APPROVED** [review]
👍 for docs

**Review by KSerrania — APPROVED** [review]

**Pull request merged**